adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -849,6 +850,15 @@ private String checkInputParameters(final String resource, final String clientId
             redirectUri = inputRedirectUri;
         }
 
+        //check the permissions required for the broker usage
+        if (AuthenticationSettings.INSTANCE.getUseBroker()) {
+            try {
+                mBrokerProxy.verifyManifestPermissions();
+            } catch (final UsageAuthenticationException exception) {
+                callback.onError(exception);
```

Comment:
UsageAuthenticationException is not a runtime exception. If we throw here directly, we have to add the throw statement in the checkInputParameters() function, which will affect all the functions calling it. That is we have to deal with the UsageAuthenticationException thrown by checkInputParameters in all the acquireToken function.